### DISCUSSION TITLE
Train fintabnet datasets

### DISCUSSION THREAD

**Discussion by fiqihfathor** [deepdoctection/deepdoctection#180] [Q&A] [answered]
i want to train fintabnet dataset in google colab, but i got an error
RuntimeError: context has already been set. How to handle it? The following is the code and error obtained

import os
import deepdoctection as dd
from deepdoctection.utils import get_configs_dir_path
from deepdoctection.train import train_faster_rcnn

fintabnet = dd.get_dataset("fintabnet")
fintabnet.dataflow.categories.filter_categories(categories="CELL")

path_config_yaml=os.path.join(get_configs_dir_path(),"dd/tp/conf_frcnn_cell.yaml")
path_weights = ""

dataset_train = fintabnet
config_overwrite=["TRAIN.STEPS_PER_EPOCH=500","TRAIN.STARTING_EPOCH=1",
                  "TRAIN.CHECKPOINT_PERIOD=50","BACKBONE.FREEZE_AT=0", "PREPROC.TRAIN_SHORT_EDGE_SIZE=[200,600]"]
build_train_config=["max_datapoints=500000"]
dataset_val = fintabnet
build_val_config = ["max_datapoints=4000"]

coco_metric = dd.metric_registry.get("coco")
coco_metric.set_params(max_detections=[50,200,600], area_range=[[0,1000000],[0,200],[200,800],[800,1000000]])

train_faster_rcnn(path_config_yaml=path_config_yaml,
                  dataset_train=dataset_train,
                  path_weights=path_weights,
                  config_overwrite=config_overwrite,
                  log_dir="/content/train_log",
                  build_train_config=build_train_config,
                  dataset_val=dataset_val,
                  build_val_config=build_val_config,
                  metric=coco_metric,
                  pipeline_component_name="ImageLayoutService"
                  )
![image](https://github.com/deepdoctection/deepdoctection/assets/66263858/5b25d6fe-db1a-4ed2-bced-8d43aa2a023f)

**Comment by JaMe76** [comment, marked as answer]
It seems at some point multiple processes get spawned. Set ` use_multi_proc=False` in ` build_train_config` and `build_val_config`

**Reply by fiqihfathor** [reply]
Thanks, i will try

**Reply by fiqihfathor** [reply]
sorry, I'm still getting the same problem. What config is meant like this?
![image](https://github.com/deepdoctection/deepdoctection/assets/66263858/0e54c0ea-5801-4b7a-9f36-83538220682a)

**Reply by fiqihfathor** [reply]
thanks, I have found the solution. With the addition of if __name__="__main__": and using a file with the py extension
